.ci/cico_build_deploy_test_rhche.sh

```diff
@@ -0,0 +1,166 @@
+# Copyright (c) 2018 Red Hat, Inc.
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Eclipse Public License v1.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v10.html
+
+# ERROR CODES:
+# 1 - missing file
+# 2 - missing credentials
+# 3 - OpenShift login failed
+# 4 - command execution failed
+
+set -x
+set -e
+set +o nounset
+
+/usr/sbin/setenforce 0
+
+export BASEDIR=$(pwd)
+export DEV_CLUSTER_URL=https://devtools-dev.ext.devshift.net:8443/
+export OC_VERSION=3.9.33
+export TARGET=${TARGET:-"rhel"}
+
+function BuildTagAndPushDocker() {
+  echo "Docker status:"
+  docker images
+  .ci/cico_build.sh
+  echo "After build:"
+  docker images
+}
+
+# Retrieve and test credentials
+set +x
+eval "$(./env-toolkit load -f jenkins-env.json -r \
+        ^DEVSHIFT_TAG_LEN$ \
+        ^QUAY_ \
+        ^KEYCLOAK \
+        ^BUILD_NUMBER$ \
+        ^JOB_NAME$ \
+        ^ghprbPullId$ \
+        ^RH_CHE)"
+
+if [[ "${PR_CHECK_BUILD}" == "true" ]]; then
+	echo "Running ${JOB_NAME} PR: #${ghprbPullId}, build number #${BUILD_NUMBER}, testing creds:"
+else
+	echo "Running ${JOB_NAME} with build, deploy and test."
+fi
+
+
+CREDS_NOT_SET="false"
+curl -s "https://mirror.openshift.com/pub/openshift-v3/clients/${OC_VERSION}/linux/oc.tar.gz" | tar xvz -C /usr/local/bin
+
+if [[ -z "${QUAY_USERNAME}" && -z "${QUAY_PASSWORD}" ]]; then
+  echo "Docker registry credentials not set"
+  CREDS_NOT_SET="true"
+fi
+
+if [[ -z "${RH_CHE_AUTOMATION_DEV_CLUSTER_SA_TOKEN}" ]]; then
+  echo "RDU2C credentials not set"
+  CREDS_NOT_SET="true"
+fi
+
+if [[ -z "${KEYCLOAK_TOKEN}" || -z "${RH_CHE_AUTOMATION_CHE_PREVIEW_USERNAME}" || -z "${RH_CHE_AUTOMATION_CHE_PREVIEW_PASSWORD}" ]]; then
+  echo "Prod-preview credentials not set."
+  CREDS_NOT_SET="true"
+fi
+
+if [ "${CREDS_NOT_SET}" = "true" ]; then
+  echo "Failed to parse jenkins secure store credentials"
+  exit 2
+else
+  echo "Credentials set successfully."
+fi
+
+if oc login ${DEV_CLUSTER_URL} --insecure-skip-tls-verify \
+                               --token "${RH_CHE_AUTOMATION_DEV_CLUSTER_SA_TOKEN}";
+then
+  echo "OpenShift login successful"
+else
+  echo "OpenShift login failed"
+  exit 3
+fi
+
+set -x
+
+# Getting core repos ready
+yum install --assumeyes epel-release
+yum update --assumeyes
+yum install --assumeyes python-pip
+
+# Test and show version
+pip -V
+
+# Getting dependencies ready
+yum install --assumeyes \
+            docker \
+            git \
+            patch \
+            pcp \
+            bzip2 \
+            golang \
+            make \
+            jq \
+            java-1.8.0-openjdk \
+            java-1.8.0-openjdk-devel \
+            centos-release-scl
+
+yum install --assumeyes \
+            rh-maven33 \
+            rh-nodejs4
+
+systemctl start docker
+pip install yq
+
+set +x
+#change version of used che
+if [[ "${USE_CHE_LATEST_SNAPSHOT}" == "true" ]]; then 
```

Comment:
If we have a job that changes the version of upstream Che it should call a specific script (this one). We can have all the custom logic here and then execute only common logic with a default script. 
In that case, the default script would be simpler (less if clauses) and this script would encapsulate all the information about the change it brings. If some info, such as image tag or namespace is needed in the default script we can pass all the info using env vars.